src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/controllers/web/WorkflowRunController.kt

```diff
@@ -44,25 +44,37 @@ class WorkflowRunController(
     fun getWorkflowRunSummaries(): List<WorkflowRunSummary>
     {
         return workflowRunRepository.getWorkflowRunSummaries(
-            context.queryParams("email"),
-            context.queryParams("namePrefix")
+                context.queryParams("email"),
+                context.queryParams("namePrefix")
         )
     }
 
-    fun getWorkflowRunSummary(): String
+    fun getWorkflowSummary(): WorkflowSummary
     {
         val response = orderlyServerAPI.post(
-            "/v1/workflow/summary/",
-            context.getRequestBody(),
-            emptyMap()
+                "/v1/workflow/summary/",
+                context.getRequestBody(),
+                emptyMap()
         )
-        return passThroughResponse(response)
+        val summary = response.data(WorkflowSummary::class.java)
+        val reports = summary.reports.map { report ->
+            val defaultParams = orderlyServerAPI.getReportParameters(report.name, context.queryParams())
+                    .associate { it.name to it.value }
+            report.copy(
+                    defaultParamList = report.params?.filter { defaultParams[it.key] == it.value }?.map {
+                        Parameter(it.key, it.value)
+                    },
+                    paramList = report.params?.filter { defaultParams[it.key] != it.value }?.map {
+                        Parameter(it.key, it.value)
+                    })
+        }
+        return WorkflowSummary(reports, summary.ref, summary.missingDependencies)
```

Comment:
It's hard to see what exactly was removed from the frontend of this PR without looking at your individual commits, but this looks to be a much more concise and elegant approach than what existed before.

Reply:
Great, glad you agree with the approach!